Add windowsevent stage loki process (#2545)
grafana/alloy: docs/sources/reference/components/loki/loki.process.md

### stage.windowsevent block

The `windowsevent` stage extracts data from the message string that appears in the Windows Event Log.

The following arguments are supported:

| Name                  | Type     | Description                                            | Default   | Required |
|-----------------------|----------|--------------------------------------------------------|-----------|----------|
| `source`              | `string` | Name of the field in the extracted data to parse.      | `message` | no       |
| `overwrite_existing`  | `bool`   | Whether to overwrite existing extracted data fields.   | `false`   | no       |
| `drop_invalid_labels` | `bool`   | Whether to drop fields that are not valid label names. | `false`   | no       |

When `overwrite_existing` is set to `true`, the stage overwrites existing extracted data fields with the same name.
If set to `false`, the `_extracted` suffix will be appended to an already existing field name.

When `drop_invalid_labels` is set to `true`, the stage drops fields that are not valid label names.

> If set to `false`, the stage will automatically convert them into valid labels replacing invalid characters with underscores.

Reviewer: ```suggestion
If set to `false`, the stage will automatically convert them into valid labels, replacing invalid characters with underscores.
```

> The `windowsevent` stage expects the message to structured in sections that are split by empty lines.

Reviewer: I'm not sure what this is trying to state.

Author: oops I forgot the "be". What I want to say is that the message in the windows event should contain sections with empty lines in between

Reviewer: ```suggestion
The `windowsevent` stage expects the message to be structured in sections that are split by empty lines.
```

The first section of the input is treated as a whole block and stored in the extracted map with the key `Description`.

> Sections following the Description are expected to contain key-value pairs in the format key:value.

Reviewer: What happens if they don't have key:value pairs? The "are expected" suggests that it may not happen this way.

Is this a "must"?

Author: Maybe we should rephrase because it's not a must but the lines that don't have a key:value pair will be ignored (unless there was already a key:value pair in the section. In this case it's considered a multi line value)

Reviewer: Maybe just adding the but about what happens if a key:value pair isn't found will help clarify what happens.

Author: It's documented a bit below with "Lines in a section without a preceding valid entry (key-value pair) are ignored and discarded."

Reviewer: Ah, got it, I was reading each line in isolation and not connecting the dots. We can just leave it as-is.

> If the first line of a section has no value (e.g., "Subject:"), the key will act as a prefix for subsequent keys in the same section.

Reviewer: ```suggestion
If the first line of a section has no value, for example "Subject:", the key will act as a prefix for subsequent keys in the same section.
```
